department-of-veterans-affairs/va.gov-cms
# VATEAM-90652: Create List & Loop pattern

## Description

Creates a Paragraph type in Drupal for the "List & Loop" pattern. (See the [Multiple responses](https://design.va.gov/patterns/ask-users-for/multiple-responses) documentation for more details on the pattern.) The Paragraph type includes an "Optional?" boolean field. Adds the new Paragraph type to the list of available Paragraph types in the "Steps" field of the Digital Form content type.

- Closes department-of-veterans-affairs/va.gov-team#90652

## Testing done
- Manually created a Digital Form with a Step that included the "List & Loop" Paragraph type.
- Ran the updated script in my local environment and verified the creation of Digital Forms with the proper Steps.

## Screenshots
The "List & Loop" pattern as part of the "Add Step" modal:
<img width="389" alt="Screenshot 2024-11-06 at 2 46 22 PM" src="https://github.com/user-attachments/assets/3875d227-b246-46ba-a8fb-0572f649560d">

An expanded "List & Loop" Paragraph inside the "Steps" field of a Digital Form:
<img width="852" alt="Screenshot 2024-11-06 at 2 47 56 PM" src="https://github.com/user-attachments/assets/dae0f1c6-cf24-4bc8-a539-01e6cbdb2565">

## QA steps

As an administrator
1. Go to Manage > Content > Add content > Digital Form
2. In the "Steps" field, click "Add Step"
   - [ ] Verify "Digital Form: List & Loop" appears as one of the patterns in the "Add Step" modal.
3. Click "Add" underneath "Digital Form: List & Loop"
   - [ ] Verify the "Digital Form: List & Loop" Paragraph appears expanded in the "Steps" field.
   - [ ] Verify the "Title" field is required and has a default value of "Multiple responses"
   - [ ] Verify a checkbox field is present labeled "Optional?"
4. Select the "Optional?" checkbox
5. Fill out the other required fields on the Digital Form and click "Save" at the bottom of the page.
   - [ ] Verify the "Multiple responses" Paragraph is included in the "Steps" field of the view page.
   - [ ] Verify the "Optional?" field is set to "Yes"

### Definition of Done


### Select Team for PR review



### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
